src/App.js

```diff
@@ -14,11 +16,114 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.PureComponent {
+  state = {
+    hideButton: false,
+    showList: true,
```

Comment:
Let's refactor this a little bit and don't use 'hidden' attribute. Let's use conditions rendering instead.
I mean that initially 'showList' should be 'false' and you should render your button according to that variable:
`{!showList && <button>Start</button>}`

by clicking this button you change your state's 'showList' to 'true' what will automatically hide the button and according to this variable render everything else:

```
{showList && (
  <GoodsList />
  ...
  <div>...</div>
  ...
  <button> ...</button>
  **and everything else here**
)}
```